test/ex_doc/formatter/html/templates_test.exs

```diff
@@ -497,13 +506,13 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
           end
         )
 
-      doc = EasyHTML.parse!(content)
-      assert doc["#example-functions a[href='#example-functions']"]
-      assert doc["#legacy a[href='#legacy']"]
-      assert doc["#example-functions [id='example/2']"]
-      refute doc["#legacy [id='example/2']"]
-      assert doc["#functions [id='example_1/0']"]
-      refute doc["#functions [id='example/2']"]
+      doc = LazyHTML.from_document(content)
+      assert num_nodes_matched(doc, "#example-functions a[href='#example-functions']") == 1
+      assert num_nodes_matched(doc, "#legacy a[href='#legacy']") == 1
+      assert num_nodes_matched(doc, "#example-functions [id='example/2']") == 1
+      assert num_nodes_matched(doc, "#legacy [id='example/2']") == 0
+      assert num_nodes_matched(doc, "#functions [id='example_1/0']") == 1
+      assert num_nodes_matched(doc, "#functions [id='example/2']") == 0
```

Comment:
`LazyHTML` implements `Access` behaviour that simply does `query` and it implements `Enumerable`, so you can get the number of top-level nodes directly. So one way would be this:

```suggestion
      assert Enum.count(doc["#example-functions a[href='#example-functions']"]) == 1
      assert Enum.count(doc["#legacy a[href='#legacy']"]) == 1
      assert Enum.count(doc["#example-functions [id='example/2']"]) == 1
      assert Enum.count(doc["#legacy [id='example/2']"]) == 0
      assert Enum.count(doc["#functions [id='example_1/0']"]) == 1
      assert Enum.count(doc["#functions [id='example/2']"]) == 0
```

Reply:
Ah - thanks! Examples are given in LazyHTML's docs, too, which I managed to not pick up on.